### PR TITLE
Rebase - JWT authentication and other fixes

### PR DESCRIPTION
**Pull request by BilalAlam173**

### CODE REVIEW

File: src/app/auth.guard.ts [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+@Injectable()
+export class AuthGuard implements CanActivate {
+
+  constructor(private UserService: UserService) {
+  }
+
+  canActivate(
+    next: ActivatedRouteSnapshot,
+    state: RouterStateSnapshot): Observable<boolean> | Promise<boolean> | boolean {
+      if ( this.UserService.getUser()) {
```
Comment (AhsanAyaz) [inline src/app/auth.guard.ts]: Could be simplified to:

```
return this.UserService.getUser();
```

also, when naming variable / properties of a class, make sure to use `camelCase`. 
I.e. `this.UserService` should actually be `this.userService.getUser()`

File: src/app/login/login.component.ts [hunk 2]
```diff
@@ -11,21 +12,22 @@ export class LoginComponent implements OnInit {
 
   email;
   password;
+  errorMessage;
 
-  constructor(private  UserService: UserService) { }
+  constructor(private  UserService: UserService,private Router: Router) { }
```
Comment (AhsanAyaz) [inline src/app/login/login.component.ts]: make sure to name the properties as `camelCase` as mentioned above.

File: src/app/login/login.component.html [hunk 3]
```diff
@@ -25,6 +25,10 @@
             </div>
             <div class="col-md-4">
                 <input type="submit" id="btn-login" class="btn btn-custom btn-lg btn-block" value="Log in" [disabled]="!loginForm.form.valid">
+                <br>
+                <a href="#signup" id="btn-login" class="btn btn-custom btn-lg btn-block btn-success">sign up</a>
```
Comment (AhsanAyaz) [inline src/app/login/login.component.html]: Any particular reason of not using `routerLink` here for redirection of routes?
If not, then I'd recommend using `routerLink` for page routing.

File: src/app/login/login.component.ts [hunk 4]
```diff
@@ -11,21 +12,22 @@ export class LoginComponent implements OnInit {
...
-  login(){
-    console.log("email:"+this.email+"-pass:"+this.password);
-    // this.UserService.login(this.email,this.password).then(function(response){
-    //   console.log(response);
-    // });
-    this.UserService.login(this.email,this.password).subscribe( data => {
-      console.log(data);
+  login() {
+    this.UserService.login(this.email, this.password).subscribe( data => {
+      if(data.success) {
```
Comment (AhsanAyaz) [inline src/app/login/login.component.ts]: Please indent the code correctly.

**Comment by BilalAlam173** [comment]
@afirdousi rebase done. sorry, if it took me longer to rebase.

File: src/app/signup/signup.component.ts [hunk 5]
```diff
@@ -27,16 +28,17 @@ export class SignupComponent implements OnInit {
     password: ''
   };
 
-  
-  constructor(private UserService: UserService) { }
+  constructor(private UserService: UserService, private Router: Router) { }
 
   ngOnInit() {
     console.log(this.User);
```
Comment (AhsanAyaz) [inline src/app/signup/signup.component.ts]: When you're done with the work. Remove the `console.log`(s)

**Review by AhsanAyaz — CHANGES_REQUESTED** [review]

File: src/app/auth.guard.ts (hunk 1 above)
Comment (BilalAlam173) [reply src/app/auth.guard.ts]: oops....

**Comment by BilalAlam173** [comment]
@AhsanAyaz done!

**Review by AhsanAyaz — APPROVED** [review]
LGTM!

**Pull request merged**